test/gtest/deepbench_rnn.cpp

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include <miopen/env.hpp>
+#include "../rnn_vanilla.hpp"
+#include "get_handle.hpp"
+
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_DEEPBENCH)
+
+namespace deepbench_rnn {
+static bool SkipTest(void)
+{
+    return (miopen::IsDisabled(ENV(MIOPEN_TEST_DEEPBENCH)) ||
+            !miopen::IsEnabled(ENV(MIOPEN_TEST_DEEPBENCH)));
+}
```

Comment:
@xinlipn BTW this is identical to `return !miopen::IsEnabled(ENV(MIOPEN_TEST_DEEPBENCH));`:

  | variable is TRUE | variable is FALSE | variable is UNSET
-- | -- | -- | --
IsDisabled returns | 0 | 1 | 0
IsEnabled returns | 1 | 0 | 0
!IsEnabled | 0 | 1 | 1
IsDisabled \|\| !IsEnabled | 0 | 1 | 1

How this should be actually implemented: see https://github.com/ROCm/MIOpen/pull/2550#discussion_r1454994117

Reply:
@atamazov , I believe CI is experiencing some issues right now. Could you help review the changes in parallel? Thanks

CC: @reidkwja , not sure if you could help take a look at the CI since local build passes. Thanks